src/example-types/metricGroups/pivot.js

```diff
@@ -31,15 +31,82 @@ let aggsForValues = (node, schema) =>
     }))
   )(node)
 
-let maybeWrapWithFilterAgg = ({ query, filters, skipFilters, aggName }) =>
-  _.isEmpty(filters) && _.isEmpty(skipFilters)
+let getKey = x => x.keyAsString || x.key
+
+let drilldownLookup = (type, path, results) => {
+  if (_.isEmpty(path) || !results) return results
+
+  let key = _.first(path)
+  let groups = _.get(type, results)
+  let match = _.find(node => getKey(node) === key, groups)
+  return drilldownLookup(type, path.slice(1), match)
+}
+
+let getResultValues = (node, results) => {
+  let pagination = node.pagination
+  let isDrilldown = _.flow(
+    _.flatMapDeep(type =>
+      _.map(prop => _.get([type, prop], pagination), ['drilldown', 'skip'])
+    ),
+    _.some(_.negate(_.isEmpty))
+  )(['columns', 'rows'])
+
+  if (!isDrilldown) return []
+
+  let groupType =
+    _.isEmpty(pagination.rows.drilldown) && _.isEmpty(pagination.rows.skip)
+      ? 'columns'
+      : 'rows'
+
+  let drilldown = _.get([groupType, 'drilldown'], pagination)
+
+  let resultNode = drilldownLookup(groupType, drilldown, results)
+  return _.map(getKey, _.get(groupType, resultNode))
+}
+
+let maybeWrapWithFilterAgg = ({
+  query,
+  filters,
+  includeRowFilters,
+  includeColumnFilters,
+  skipFilters,
+  aggName,
+}) =>
+  _.isEmpty(filters) &&
+  _.isEmpty(includeRowFilters) &&
+  _.isEmpty(includeColumnFilters) &&
+  _.isEmpty(skipFilters)
     ? query
     : {
         aggs: {
           [aggName]: {
             filter: {
               bool: {
-                ...(!_.isEmpty(filters) && { must: filters }),
+                ...((!_.isEmpty(filters) ||
+                  !_.isEmpty(includeRowFilters) ||
+                  !_.isEmpty(includeColumnFilters)) && {
+                  must: [
+                    ...(filters || []),
+                    ...(!_.isEmpty(includeRowFilters)
+                      ? [
+                          {
+                            bool: {
+                              should: includeRowFilters,
+                            },
+                          },
+                        ]
+                      : []),
+                    ...(!_.isEmpty(includeColumnFilters)
+                      ? [
+                          {
+                            bool: {
+                              should: includeColumnFilters,
+                            },
+                          },
+                        ]
+                      : []),
+                  ],
+                }),
```

Comment:
This query structure doesn't seem right - if you're representing the row/column filters as things `or`ed together, you probably want `minimum_should_match` set to one. With that said, you might be able to simplify this if these aren't meant to be `or`ed. We definitely need test cases covering these scenarios and some comments explaining the reasoning.

Reply:
Yeah, those are meant to be `or`ed. It's like matching one of these 10 rows and one of these 5 columns simultaneously.

Reply:
Added comments as well